codice/ddf
# DDF-3719 UI tooling updates

#### What does this PR do?

- Updates `ace start` to always open a browser, but reuse browser tabs
  when available on osx, see `react-dev-utils/openBrowser`.
  - Adds `-N, --no-open` flag to prevent this behavior.
- Adds `--proxy` flag to `ace start` to allow users to specify backend
  url.
- Adds prestart script to catalog-ui-search which tries to disable
  IDP authentication before starting the dev server. If it fails, it
  will display a warning and continue starting the dev server.
- Updates `ace test` to report exit code of phantom sub-process. Tests
  were failing silently before.
- Adds `yarn m2` command to push artifacts into `~/.m2` for
  root and catalog-ui-search packages.
- Updates dev server to proxy `/search` instead of `/`.
  - This causes issues with dev server's own endpoints.
- Removes unused test scripts in admin-federation-ui
- Fixes keyword spec failure in catalog-ui-search package.




#### Who is reviewing it? 


@adimka 
@garrettfreibott 

#### Ask 2 committers to review/merge the PR and tag them here.


@andrewkfiedler 
@brendan-hofmann 

#### How should this be tested?


- `mvn clean install` the ui directory
- start the dev server
  - ensure you have no issues related to IDP/CSRF
  - ensure you can proxy to another ddf address besides the default
    - you should try with a positive and a negative test
- update a spec in catalog-ui-search to fail and ensure it fails the build

#### What are the relevant tickets?

[DDF-3719](https://codice.atlassian.net/browse/DDF-3719)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
